### PR TITLE
DATACMNS-988 - Provide RxJava 2 Repository interfaces.

### PR DESCRIPTION
**Pull request by mp911de**
Provide RxJava 2 Repository interfaces using back-pressure enabled `Flowable` as default return type for multi-element queries and `Maybe` for queries that can complete without emitting a value.

---

Related ticket: [DATACMNS-988](https://jira.spring.io/browse/DATACMNS-988).

### CODE REVIEW

File: src/main/java/org/springframework/data/repository/reactive/RxJava2CrudRepository.java [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+
+/**
+ * Interface for generic CRUD operations on a repository for a specific type. This repository follows reactive paradigms
+ * and uses RxJava 2 types.
+ *
+ * @author Mark Paluch
+ * @since 2.0
+ * @see Single
+ * @see Flowable
+ * @see Completable
```
Comment (simonbasle) [inline src/main/java/org/springframework/data/repository/reactive/RxJava2CrudRepository.java]: Maybe add `@see Maybe` here too :p

File: src/main/java/org/springframework/data/repository/reactive/RxJava2CrudRepository.java [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+	 * Returns whether an entity with the given id exists.
+	 *
+	 * @param id must not be {@literal null}.
+	 * @return {@literal true} if an entity with the given id exists, {@literal false} otherwise.
+	 * @throws IllegalArgumentException if {@code id} is {@literal null}.
+	 */
+	Single<Boolean> exists(ID id);
+
+	/**
+	 * Returns whether an entity with the given id exists supplied by a {@link Single}.
```
Comment (simonbasle) [inline src/main/java/org/springframework/data/repository/reactive/RxJava2CrudRepository.java]: rewording as `Returns whether an entity with the given id, supplied by a {@link Single}, exists` would be a tiny bit clearer IMHO

File: src/main/java/org/springframework/data/repository/util/ReactiveWrapperConverters.java [hunk 3]
```diff
@@ -722,7 +722,7 @@ public Completable convert(Publisher<?> source) {
 
 		@Override
 		public io.reactivex.Observable<?> convert(Publisher<?> source) {
-			return (io.reactivex.Observable<?>) REACTIVE_ADAPTER_REGISTRY.getAdapter(io.reactivex.Single.class)
+			return (io.reactivex.Observable<?>) REACTIVE_ADAPTER_REGISTRY.getAdapter(io.reactivex.Observable.class)
```
Comment (simonbasle) [inline src/main/java/org/springframework/data/repository/util/ReactiveWrapperConverters.java]: shouldn't it be `Flowable`?

File: src/test/java/org/springframework/data/repository/util/ReactiveWrapperConvertersUnitTests.java [hunk 4]
```diff
@@ -175,6 +175,14 @@ public void toWrapperShouldConvertRxJava2ObservableToPublisher() {
 		assertThat(ReactiveWrapperConverters.toWrapper(foo, Publisher.class)).isInstanceOf(Publisher.class);
 	}
 
+	@Test // DATACMNS-988
+	public void toWrapperShouldConvertPublusherToRxJava2Observable() {
```
Comment (simonbasle) [inline src/test/java/org/springframework/data/repository/util/ReactiveWrapperConvertersUnitTests.java]: shouldn't it be `Flowable`?

File: src/main/java/org/springframework/data/repository/util/ReactiveWrapperConverters.java (hunk 3 above)
Comment (mp911de) [inline src/main/java/org/springframework/data/repository/util/ReactiveWrapperConverters.java]: That's the converter to adopt a `Publisher` to an `Observable`. `Flowable` adoption happens in the [`PublisherToRxJava2FlowableConverter `](https://github.com/spring-projects/spring-data-commons/blob/2.0.x/src/main/java/org/springframework/data/repository/util/ReactiveWrapperConverters.java#L864). That's a fix for a bug we didn't noticed earlier.

Comment (simonbasle) [inline src/main/java/org/springframework/data/repository/util/ReactiveWrapperConverters.java]: got it 👍

File: src/main/java/org/springframework/data/repository/reactive/RxJava1CrudRepository.java [hunk 5]
```diff
@@ -73,7 +73,7 @@
 	Observable<T> findOne(ID id);
 
 	/**
-	 * Retrieves an entity by its id supplied by a {@link Single}.
+	 * Returns whether an entity with the given id, supplied by a {@link Single}, exists.
```
Comment (simonbasle) [inline src/main/java/org/springframework/data/repository/reactive/RxJava1CrudRepository.java]: oopsie, that's the wrong method you changed the javadoc on, for `RxJava1CrudRepository` :p

**Pull request closed without merging**